### DISCUSSION TITLE
How do I pass my input from one node to the next?

### DISCUSSION THREAD

**Discussion by neil-muse** [gorules/zen#123] [Q&A]
I'm trying to understand how I pass my Input from one decision table to the next or into a function to do a calculation. If I have a value like a salary, I might run this through a couple of downstream decision tables or into a function. Could you tell me how I do this task? I tried to add the input value to my output in the table using the same selector name, but nothing came through

**Comment by ivanmiletic** [comment]
Hi @neil-muse would you be able to share more info or send us an example of your decision model? Please omit private info such as amounts or names if you have them.

In short, you can pass info from the single node to multiple nodes downstream by connecting them. Example:
![Screenshot 2024-02-14 at 22 09 42](https://github.com/gorules/zen/assets/60513195/01c7a2ca-7ecc-41bc-ada0-9953c073f11b)

Order of evaluation Request -> Table 1 -> Table 2 -> Function 1 -> Response.

Table 1 input will be Request
Table 2 input will be merged Request + Table 1 Output
Function 1 input will be merged Request + Table 1 output + Table 2 output
Response will be merged Table 2 output + Function 1 output

**Reply by neil-muse** [reply]
Thanks, @ivanmiletic, for the quick reply.

So what you are saying is I have an input request, for example 

    {
        "filing_status": "single",
        "salary": 50000,
        "boxes_checked": 0,
    }
    
If I want to reference it in downstream tables or functions, I need to connect it. In your example, I might use filing_status and box_checked in table 1 and salary and filing status in table 2. In the function 1, I would have the outputs of table 1 and table 2 plus my original request. It is not like the input request context is also passed with the outputs. At least in the simulator, I'm only seeing the final output. Or is there a way to reference the original request in tables, functions, and expressions?

**Reply by ivanmiletic** [reply]
You can connect the Request to many nodes, and you can even control what you send as output by connecting all nodes you wish to the Response.
